### DISCUSSION TITLE
Open Source Upstream Contributions

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3682] [DDEV Advisory Group]
## Context
Per the [March 9, 2022 Budget Meeting](https://github.com/drud/ddev/discussions/3681), the [LocalDev Foundation](https://localdev.foundation/) is bringing in a monthly estimated income of $1665 and the first payout should be $5800 on March 22. While the foundation is a registered not-for-profit, it is not (as of right now) tax exempt. Additionally, one of the foundation's goals is to help financially support the projects that DDEV relies upon. To this end, we're asking the DDEV community to propose projects for the foundation's financial support

## What makes a project a good candidate?
A good candidate for the foundation's support will be:
- Open source
- Community-driven
- Not solely/primarily sponsored by a corporation (e.g. VS Code by Microsoft)
- A project that directly relates to the development or direct use of DDEV (e.g. mutagen, colima, xDebug)
- A project that relates to the communities that use DDEV (e.g. projects that are more for Drupal, Wordpress, TYPO3 development)

## Projects already targeted
- [Mutagen](https://github.com/mutagen-io/mutagen)
- [xDebug](https://github.com/xdebug/xdebug)
- [deb.sury.org](https://github.com/oerdnj/deb.sury.org)
- [Homebrew](https://github.com/Homebrew)
- @shaal for [DrupalPod](https://github.com/shaal/DrupalPod)

**Comment by chrisfromredfin** [comment]
Randy I believe said we also already support https://deb.sury.org/ as we rely heavily on those repos to get our differing/dated PHP versions.

**Reply by rfay** [reply]
Yes, and should increase from the current $10/month

**Comment by deviantintegral** [comment]
I'd thought Mutagen had some sort of funding and was going down the route of productizing it. https://mutagen.io/login seems like a step towards that. Others likely know more, but it may not be a great example (and might be why they are slow in setting up GitHub sponsors?).

Should we also think about what a review process is like? For example, perhaps we sponsor a project, but after 6 or 12 months it seems unmaintained or abandoned.

I think it would also be good to do a "community health" evaluation. The software may be great, but at times we may not want the ddev community associated with an upstream maintainer or community due to toxic behaviour. It could be as simple as "do contributors and users get friendly and reasonable responses in issues" or for bigger orgs "do they have a code of conduct".

**Reply by rfay** [reply]
Great question about Mutagen - I've been pestering Jacob to finish his Github Sponsors account, but he has bigger fish to fry that are far more important. Having to do with a certain war we're all worrying about.  He does have hopes of a broader funding mechanism, but it's not there yet, and maintenance of that project is certainly a huge priority now for DDEV.

Agree about review strategy. I like the "What makes a good candidate" section.

Question about "community health" is a great one. Although deb.sury.org is a great resource for us... it's not a very friendly issue queue. But so very important.

**Comment by mglaman** [comment]
What about the PHP Foundation? https://opencollective.com/phpfoundation

**Comment by siliconmeadow** [comment]
How does one contribute to the LocalDev Foundation in the first place? I missed that bit!

**Reply by andy-blum** [reply]
For right now, you can sponsor @rfay on github. All those sponsor dollars are currently directed toward the localdev foundation

**Reply by rfay** [reply]
@andy-blum the details are at https://github.com/sponsors/rfay - sponsorships go to the Localdev Foundation. Thanks for asking!

**Reply by bmack** [reply]
good point, this should be clarified on the Localdev Foundation website?

**Reply by rfay** [reply]
@bmack I updated https://localdev.foundation, PRs are welcome also on https://github.com/rfay/localdev.foundation

**Comment by bmack** [comment]
The tools that DDEV uses (that come to my mind, regardless if there is a big corporation behind):

* MySQL
* MariaDB
* PHP
* Node
* Go
* Docker
* nginx
* Apache
* mkcert

**Reply by bmack** [reply]
On the other hand, we have the CMSes that actually "benefit" from DDEV, so I think the CMSes should actually give their 10% back TO DDEV ;) But maybe there could be partnerships to promote DDEV as the "go-to-local-dev" tool?

**Comment by rfay** [comment]
I use iterm2 loads and have for years. I think it's an important part of many of our workflows.

**Comment by rfay** [comment]
## Current monthly upstream donations ($95/month)

- PHP Foundation - Open Collective - $30/month
- Patreon - Colima - $30/month
- shaal/Drupalpod - $5/month
- oerdnj/deb.sury.org - $10/month
- derickr/xdebug - $10/month
- homebrew - $10/month
